tests/utils.py

```diff
@@ -169,12 +169,17 @@ def get_image_name_from_csv(image_string, csv_related_images):
     raise ResourceNotFoundError(f"no image with the string {image_string} was found in the csv_dict")
 
 
-def hotplug_resource_and_wait_hotplug_migration_finish(vm, client, sockets=None, memory_guest=None):
-    assert sockets or memory_guest, "No resource for update provided!!!"
-    assert not (sockets and memory_guest), "Hotpluging both cpu and memory at once is not supported!"
+def hotplug_resource_and_wait_hotplug_migration_finish(
+    vm, client, sockets=None, memory_guest=None, new_instance_type=None
+):
+    assert bool(sockets or memory_guest) ^ bool(new_instance_type), "wrong parameter values for Hotpluging"
+    if new_instance_type:
+        instance_type_spec = new_instance_type.instance.spec
+        sockets = instance_type_spec.cpu.guest
+        memory_guest = instance_type_spec.memory.guest
```

Comment:
this function becomes pretty complicated to read, it would be better to have 2 separate functions for regular VMs and VMs from instance type

for the common part you can create a supportive function that will check migration and updated values

Reply:
I split the functions in another PR - https://github.com/RedHatQE/openshift-virtualization-tests/pull/824
Moving this to hold until it is merged